apps/daimo-mobile/src/view/screen/history/shared.ts

```diff
@@ -11,34 +11,58 @@ import { env } from "../../../env";
 // Either uses the memo field for standard transfers, e.g. "for ice cream"
 // Or generates a synthetic one for swaps, e.g. "5 USDT -> USDC" if short
 // or "Accepted 5 USDT as USDC" if long
-export function getSynthesizedMemo(
-  op: DisplayOpEvent,
-  daimoChain: DaimoChain,
-  short?: boolean
-) {
+export function getSynthesizedMemo({
+  op,
+  daimoChain,
+  short,
+  sentByUs,
+}: {
+  op: TransferClog;
+  daimoChain: DaimoChain;
+  short?: boolean;
+  sentByUs?: boolean;
+}) {
   const chainConfig = env(daimoChain).chainConfig;
   const coinName = chainConfig.tokenSymbol.toUpperCase();
 
   if (op.memo) return op.memo;
   if (op.type === "createLink" && op.noteStatus.memo) return op.noteStatus.memo;
   if (op.type === "claimLink" && op.noteStatus.memo) return op.noteStatus.memo;
 
-  if (op.type !== "transfer") return null;
-  if (op.requestStatus) {
+  if (op.type !== "transfer" && op.type !== "swap") return null;
+  if (op.type === "transfer" && op.requestStatus) {
     return op.requestStatus.memo;
-  } else if (op.preSwapTransfer) {
-    if (isNativeETH(op.preSwapTransfer.coin, chainConfig)) {
-      return `ETH → ${coinName}`;
+  } else if (
+    (op.type === "transfer" && op.preSwapTransfer) ||
+    op.type === "swap"
+  ) {
+    const isOutboundSwap = !!(op.type === "swap" && sentByUs);
```

Comment:
this logic is a bit involved. i recommend splitting out the transfer and swap cases

in fact, the most clear might be this:

```ts
switch (op.type) {
  case "createLink":
    return op.noteStatus.memo || null;
  // ...
  case "transfer":
    return getMemoFromTransfer(op);
  // ...
}
```